Assets/Scripts/SS3D/Systems/Health/StaminaController.cs

```diff
@@ -0,0 +1,153 @@
+using FishNet.Object;
+using FishNet.Object.Synchronizing;
+using SS3D.Core.Behaviours;
+using SS3D.Systems.Entities;
+using SS3D.Systems.Entities.Humanoid;
+using SS3D.Systems.Health.UI;
+using System;
+using UnityEngine;
+
+namespace SS3D.Systems.Health
+{
+    /// <summary>
+    /// Manages the networking and out-of-process dependencies of the stamina subsystem
+    /// </summary>
+    [RequireComponent(typeof(HumanoidController))]
+    public class StaminaController : NetworkActor
+    {
+        /// <summary>
+        /// The stamina bar UI
+        /// </summary>
+        private StaminaBarView UI;
+
+        /// <summary>
+        /// The type of stamina data to create
+        /// </summary>
+        [SerializeField] private StaminaType _staminaType;
+
+        /// <summary>
+        /// The controller for this entity.
+        /// </summary>
+        [SerializeField] private HumanoidController _player;
+
+
+        [SerializeField] private PlayerControllable playerControllable;
+
+        /// <summary>
+        /// Provides a way for the client to access the current player stamina.
+        /// </summary>
+        [SyncVar] private float _current;
+
+        public float Current
+        {
+            get { return _current; }
+        }
+
+        /// <summary>
+        /// Actual stamina data. Will only exist on the server.
+        /// </summary>
+        private IStamina _stamina;
+
+        protected override void OnStart()
+        {
+            base.OnStart();
+
+            // The server manages the stamina data for each entity.
+            if (IsServer)
+            {
+                _stamina = StaminaHelper.Create(_staminaType, 10f);
+                _current = _stamina.Current;
+            }
+
+            // Currently movement is client-authoritative, so we need to subscribe to events on the client only.
+            if (IsClient && playerControllable.ControllingSoul.IsOwner)
+            {
+                SubscribeToEvents();
```

Comment:
I don't think this will work when switching bodies. You could listen to `ControllingSoulChanged` on `playerControllable`

Reply:
Thanks, I have reworked that and tested with MindSwap. It seems to work properly now.